service/src/main/java/bio/terra/profile/service/profile/ProfileService.java

```diff
@@ -167,6 +162,32 @@ public List<ProfileDescription> listProfiles(
     }
   }
 
+  public ProfileModel updateProfile(
+      UUID id, UpdateProfileRequest requestBody, AuthenticatedUserRequest user) {
+    BillingProfile profile = getProfileWithAccessCheck(id, user);
+    String description = String.format("Update billing profile id [%s]", id);
+    logger.info(description);
+    var updateJob =
+        jobService
+            .newJob()
+            .description(description)
+            .flightClass(UpdateProfileFlight.class)
+            .request(requestBody)
+            .addParameter(ProfileMapKeys.PROFILE, profile)
+            .userRequest(user);
+    return updateJob.submitAndWait(BillingProfile.class).toApiProfileModel();
+  }
+
+  public void removeBillingAccount(UUID id, AuthenticatedUserRequest user) {
```

Comment:
Why does the normal profile update require stairway, but `removeBillingAccount` doesn't?

On a related note, I wonder if we might be over-using stairway, both here, and maybe even more generally. 
I'm not sure we really need the multi-step, all-or-nothing, unwindable guarantees when there's only a single change being made, and it's to this service's db. Unless I'm missing something, the rest of the steps are just validations and access checks, which there's no reason to run in stairway. 

I guess there's the separate part of returning the updated object, but we could either not do that, or find some other way that doesn't require a separate db call (looks like `ProfileDao.createBillingProfile()` already just returns the pojo from memory, instead of running a separate query).

Reply:
Yeah, I went back and forth on whether to make this a flight or not. Ultimately, I landed on the flight when introducing the GCP authz check as the external cloud call pushed me over the edge. It does give us a bit more flexibility going forward in case we need more complex profile updates, but I wouldn't say that alone is worth the added complexity of a flight.

As for your original question, a single authz check with Sam followed by the update didn't feel worth a flight. Perhaps that's telling enough for the original flight question -- adding two authz checks are probably not enough to warrant a flight